### ISSUE TITLE
OptionalWhenBraces: false positive with lambda which has no arrow

### ISSUE THREAD

**Issue by t-kameyama** [arturbosch/detekt#2567]
## Steps to Reproduce
```kotlin
fun test(b: Boolean): (Int) -> Int {
    return when (b) {
        true -> { { it + 100 } }
        false -> { { it + 200  } }
    }
}
```
